kingpin/actors/aws/ecs.py

```diff
@@ -406,7 +420,369 @@ def _execute(self):
+        'tokens': (dict, {},
+                   'A dictionary of key/value pairs '
+                   'used to fill in the tokens for the Task and Service '
+                   'definition templates.'),
+        'count': (int, 1, 'How many instances of the service to deploy.'),
+        'wait': (bool, True,
+                 'Whether to wait for the services to deploy.')
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(DeployService, self).__init__(*args, **kwargs)
+        self.task_definition = _load_task_definition(
+            self.option('task_definition'),
+            self.option('tokens'))
+        self.service_definition = _load_service_definition(
+            self.option('service_definition'),
+            self.option('tokens'))
+
+    def _get_service_name(self, family):
+        """Gets service_name from either option 'service_name' or given family.
+
+        Args:
+            family: service_name to fallback to if 'service_name' was not set.
+
+        Returns:
+            service_name to use.
+        """
+        # Optionally use specified 'service_name' instead of task family.
+        service_name = self.option('service_name')
+        if service_name is None:
+            service_name = family
+        return service_name
+
+    @gen.coroutine
+    def _describe_service(self, service_name):
+        """Describe a service by name.
+
+        Args:
+            service_name: service name to describe.
+
+        Returns:
+            Service param dict, or None if the service does not exist.
+        """
+        response = yield self.thread(
+            self.ecs_conn.describe_services,
+            cluster=self.option('cluster'),
+            services=[service_name])
+        self._handle_failures(response['failures'], self.FAILURE_MISSING)
+
+        services = response['services']
+
+        # There should never be more than one service for a given name.
+        if len(services) > 1:
+            self.log.warning(
+                'API returned more than one service for {name}'.format(
+                    name=service_name))
+
+        if services:
+            raise gen.Return(services[0])
+
+    @gen.coroutine
+    @dry('Would create service.')
+    def _create_service(self, service_name, family, revision, client_token):
+        """Create a service.
+
+        Args:
+            service_name: service name to use.
+            family: task definition family.
+            revision: task definition revision.
+            client_token: uuid for the service creation.
+        """
+        create_parameters = dict(
+            cluster=self.option('cluster'),
+            serviceName=service_name,
+            taskDefinition='{0}:{1}'.format(family, revision),
+            desiredCount=self.option('count'),
+            clientToken=client_token,
+            **self.service_definition)
+
+        self.log.info('Creating service.')
+
+        yield self.thread(
```

Comment:
Exception catching ... what happens if there is a failure here? Can we tell the user a bit about the failure?